packages/client/hmi-client/src/components/code/tera-code.vue

```diff
@@ -0,0 +1,264 @@
+<template>
+	<main>
+		<header>
+			<section class="header">
+				<!-- <h4>{{ name }}</h4> -->
+				<section class="name">
+					<InputText v-model="codeName" class="name-input" />
+				</section>
```

Comment:
This should be a conditional if the asset as a name.

Reply:
I wanted to test something out here; I think the title (if editable) should always be an InputText. What's the point of having a separate edit button and having extra conditional logic? What do you think @jamiewaese-uncharted 

https://github.com/DARPA-ASKEM/Terarium/assets/16358728/2af113c7-76db-4a57-95dd-e95ea8563c9c

Reply:
I don't think we should do that. If a user click on it by inadvertence and type keys that get saved automatically. This will be really error prone.

Reply:
It seems very unlikely to me that a user would actually accidentally click it. If they do, anything they accidentally type is very noticeable, and the consequences are trivial and easy to undo (the name does not automatically save when you focus out of the element). 

Whereas having to click a small edit button is much more annoying, and the user will face this annoyance almost every time.

Reply:
Is the title of an asset changed often? If not, I would think we should stick with the current design of clicking an edit button to enable the input field to avoid inadvertently re-naming the assets as Yohann mentioned. Thoughts @jamiewaese-uncharted and @pascaleproulx ?

Reply:
Maybe we can use the PrimeVue `inplace` component here: https://primevue.org/inplace/#basic

It would look sorta like this, but we use a heading font size, and we add an 'X' button to cancel and a checkmark button to confirm.

![vue inplace](https://github.com/DARPA-ASKEM/Terarium/assets/120480244/ac2a0fcb-1897-45b4-8def-3e1c5c7f9526)

Reply:
Sure @echl if you could make a ticket of it.